framework/cmd/ci.go

```diff
@@ -0,0 +1,279 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"go.uber.org/ratelimit"
+	"golang.org/x/sync/errgroup"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/google/go-github/v50/github"
+	"github.com/smartcontractkit/chainlink-testing-framework/framework"
+	"golang.org/x/oauth2"
+)
+
+const (
+	WorkflowRateLimitPerSecond = 10
+	JobsRateLimitPerSecond     = 10
+	MaxBarLength               = 50
+	GHResultsPerPage           = 100 // anything above that won't work
+)
+
+var (
+	SlowTestThreshold          = 5 * time.Minute
+	ExtremelySlowTestThreshold = 10 * time.Minute
+)
+
+type JobResult struct {
+	StepStats map[string]Stat
+	JobStats  map[string]Stat
+}
+
+type Stat struct {
+	Name      string
+	Median    time.Duration
+	P95       time.Duration
+	P99       time.Duration
+	Durations []time.Duration
+}
+
+// AnalyzeCIRuns analyzes GitHub Actions job runs and prints statistics
+func AnalyzeCIRuns(owner, repo, wf string, daysRange int) error {
+	ctx := context.Background()
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		return fmt.Errorf("GITHUB_TOKEN environment variable is not set")
+	}
+
+	framework.L.Info().
+		Str("Owner", owner).
+		Str("Repo", repo).
+		Str("Workflow", wf).
+		Msg("Analyzing CI runs")
+
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
```

Comment:
maybe use `lib/client/github`?

Reply:
It's just two methods, maybe if we add more in the next PRs it'd make sense to switch to it, if not, I prefer no inter-dependencies for the framework CLI 🙏